tests/dependency-test.js

```diff
@@ -0,0 +1,116 @@
+const { assert } = require('chai');
+const getLoad = require('../dependencies');
+
+
+describe('Dependency logic', function () {
+
+	/** @type {import("../dependencies").Components} */
+	const components = {
+		languages: {
+			meta: {},
+			'a': {
+				alias: 'a2'
+			},
+			'b': {
+				alias: 'b2'
+			},
+			'c': {
+				require: 'a',
+				after: ['b', 'e']
+			},
+			'd': {
+				require: ['c', 'b'],
+				alias: 'xyz'
+			},
+		},
+		pluginsOrSomething: {
+			meta: {},
+			'e': {
+				modify: 'a'
+			},
+		}
+	};
+
+	/**
+	 * Returns the ids of `getLoad`.
+	 *
+	 * @param {string[]} load
+	 * @param {string[]} [loaded]
+	 * @returns {string[]}
+	 */
+	function getIds(load, loaded) {
+		return getLoad(components, load, loaded).ids
+	}
+
+	/**
+	 * Returns the load order of `getLoad`.
+	 *
+	 * @param {string[]} load
+	 * @param {string[]} [loaded]
+	 * @returns {string[]}
+	 */
+	function loadOrder(load, loaded) {
+		const order = [];
+		getLoad(components, load, loaded).load(id => order.push(id));
+		return order;
```

Comment:
This function would be useful exported rather than just as a test function.

Reply:
Instead, we could also replace the `ids` of the returned object with a `getIds` function which returns the ids in `ids` in order. I just want to keep the API minimal.

Also, where is this useful? The functional interface is more powerful.

Reply:
Oh, I see. Can this function be shortened to `return getLoad(...).load(id => id)`?

Reply:
No, it can't. 
I changed the API, so we now have a `getIds` function which returns all ids in topological order.